src/System.ComponentModel.TypeConverter/tests/TypeConverterTestBase.cs

```diff
@@ -0,0 +1,325 @@
+            public Type DestinationType { get; set; }
+            public object Expected { get; set; }
+            public CultureInfo Culture { get; set; }
+            public Type NetCoreExceptionType { get; set; }
+            public Type NetFrameworkExceptionType { get; set; }
+            public ITypeDescriptorContext Context { get; set; } = TypeConverterTests.s_context;
+            public CultureInfo RemoteInvokeCulture { get; set; }
+
+            public bool CanConvert { get; set; }
+            
+            public static ConvertTest Valid(object source, object expected, CultureInfo culture = null)
+            {
+                return new ConvertTest
+                {
+                    Source = source,
+                    DestinationType = expected?.GetType(),
+                    Expected = expected,
+                    Culture = culture,
+                    CanConvert = true,
+                };
+            }
+
+            public static ConvertTest Throws<TException>(object source, CultureInfo culture = null) where TException : Exception
+            {
+                return Throws<TException, TException>(source, culture);
+            }
+
+            public static ConvertTest Throws<TNetCoreException, TNetFrameworkException>(object source, CultureInfo culture = null) where TNetCoreException : Exception where TNetFrameworkException : Exception
+            {
+                return new ConvertTest
+                {
+                    Source = source,
+                    Culture = culture,
+                    NetCoreExceptionType = typeof(TNetCoreException),
+                    NetFrameworkExceptionType = typeof(TNetFrameworkException),
+                    CanConvert = true
+                };
+            }
+
+            public static ConvertTest CantConvert(object source, Type destinationType = null, CultureInfo culture = null)
+            {
+                return new ConvertTest
+                {
+                    Source = source,
+                    DestinationType = destinationType,
+                    Culture = culture,
+                    NetCoreExceptionType = typeof(NotSupportedException),
+                    NetFrameworkExceptionType = typeof(NotSupportedException),
+                    CanConvert = false
+                };
+            }
+
+            public ConvertTest WithContext(ITypeDescriptorContext context)
+            {
+                Context = context;
+                return this;
+            }
+
+            public ConvertTest WithRemoteInvokeCulture(CultureInfo culture)
+            {
+                RemoteInvokeCulture = culture;
+                return this;
+            }
+
+            public ConvertTest WithInvariantRemoteInvokeCulture() => WithRemoteInvokeCulture(CultureInfo.InvariantCulture);
+
+            public ConvertTest()
+            {
+            }
+  
+            protected ConvertTest(SerializationInfo info, StreamingContext context)
```

Comment:
All this stuff is necessary as `RemoteInvoke` can't be use locals from outside of the scope of the action method. Basically what we do is serialize the object to a base64 string using binary converter then deserialize it after. Is there a better way of doing this? I also needed to implement this as `RuntimeType` and `CultureInfo` are not serializable by default.